ui/de-lib/src/main/java/org/iplantc/de/client/models/apps/PublishAppRequest.java

```diff
@@ -0,0 +1,42 @@
+package org.iplantc.de.client.models.apps;
+
+import org.iplantc.de.client.models.HasDescription;
+import org.iplantc.de.client.models.HasId;
+import org.iplantc.de.client.models.avu.Avu;
+
+import com.google.gwt.user.client.ui.HasName;
+import com.google.web.bindery.autobean.shared.AutoBean.PropertyName;
+
+import java.util.List;
+
+/**
+ * @author aramsey
+ */
+public interface PublishAppRequest extends HasId, HasName, HasDescription {
```

Comment:
Note that the service endpoint's `PublishAppRequest` schema allows the `id`, `integration_date`, and `edited_date` fields, but these were only allowed as a convenience to the UI, which was sending an app object that included these fields to the service.

The endpoint doesn't check these fields (the ID is already included as part of the endpoint's URL), so they may be safely removed from this new `PublishAppRequest` model. In a future update, the endpoint may distinguish between app creator and app publisher, so we can update both this model and the endpoint at that time to remove these extraneous fields.